src/Avalonia.Base/Input/Platform/KeyGestureFormatInfo.cs

```diff
@@ -0,0 +1,133 @@
+        /// <param name="formatProvider">The IFormatProvider to get a KeyGestureFormatInfo for.</param>
+        /// <returns></returns>
+        public static KeyGestureFormatInfo GetInstance(IFormatProvider? formatProvider)
+            => formatProvider?.GetFormat(typeof(KeyGestureFormatInfo)) as KeyGestureFormatInfo
+            ?? AvaloniaLocator.Current.GetService<KeyGestureFormatInfo>()
+            ?? Invariant;
+
+        /// <summary>
+        /// A dictionary of the common platform Key overrides. These are used as a fallback
+        /// if platformKeyOverrides doesn't contain the Key in question.
+        /// </summary>
+        
+        private static readonly Dictionary<Key, string> s_commonKeyOverrides = new()
+        {
+            { Key.Add , "+" },
+            { Key.D0 , "0" },
+            { Key.D1 , "1" },
+            { Key.D2 , "2" },
+            { Key.D3 , "3" },
+            { Key.D4 , "4" },
+            { Key.D5 , "5" },
+            { Key.D6 , "6" },
+            { Key.D7 , "7" },
+            { Key.D8 , "8" },
+            { Key.D9 , "9" },
+            { Key.Decimal , "." },
+            { Key.Divide , "/" },
+            { Key.Multiply , "*" },
+            { Key.OemBackslash , "\\" },
+            { Key.OemCloseBrackets , "]" },
+            { Key.OemComma , "," },
+            { Key.OemMinus , "-" },
+            { Key.OemOpenBrackets , "[" },
+            { Key.OemPeriod , "." },
+            { Key.OemPipe , "|" },
+            { Key.OemPlus , "+" },
+            { Key.OemQuestion , "/" },
+            { Key.OemQuotes , "\"" },
+            { Key.OemSemicolon , ";" },
+            { Key.OemTilde , "`" },
+            { Key.Separator , "/" },
+            { Key.Subtract , "-" },
+            { Key.Back , "Backspace" },
+            { Key.Down , "Down Arrow" },
+            { Key.Left , "Left Arrow" },
+            { Key.Right , "Right Arrow" },
+            { Key.Up , "Up Arrow" }
+        };
+
+        /// <summary>
+        /// Checks the platformKeyOverrides and s_commonKeyOverrides Dictionaries, in order, for the appropriate
+        /// string to represent the given Key on this platform.
+        /// NOTE: If platformKeyOverrides is null, this is assumed to be the Invariant and the Dictionaries are not checked.
+        /// The plain Enum string is returned instead.
+        /// </summary>
+        /// <param name="key">The Key to format.</param>
+        /// <returns>The appropriate platform specific or common override if present, key.ToString() if not or this is the Invariant.</returns>
+        public string FormatKey(Key key)
+        {
+            /*
+             * The absence of an Overrides dictionary indicates this is the invariant, and
+             * so should just return the default ToString() value.
+             */
+            if (platformKeyOverrides == null)
+                return key.ToString();
+
+            return platformKeyOverrides.TryGetValue(key, out string? result) ? result : 
```

Comment:
Nit: can be simplified to `platformKeyOverrides.TryGetValue(key, out var result) || commonKeyOverrides.TryGetValue(key, out result) ? result : key.ToString()`, avoiding the nested ternary operators.